packages/@aws-cdk/aws-appmesh/lib/virtual-router-listener.ts

```diff
@@ -0,0 +1,93 @@
+import * as cdk from '@aws-cdk/core';
+import { CfnVirtualRouter } from './appmesh.generated';
+import { Protocol } from './shared-interfaces';
+
+/**
+ * Properties for a VirtualRouter listener
+ */
+export interface VirtualRouterListenerConfig {
+  /**
+   * Single listener config for a VirtualRouter
+   */
+  readonly listener: CfnVirtualRouter.VirtualRouterListenerProperty;
+}
+
+/**
+ * Represents the properties needed to define listeners for a VirtualRouter
+ */
+export abstract class VirtualRouterListener {
+  /**
+   * Returns an HTTP Listener for a VirtualRouter
+   */
+  public static http(port?: number): VirtualRouterListener {
+    return new VirtualRouterListenerImpl(Protocol.HTTP, port);
+  }
+
+  /**
+   * Returns an HTTP2 Listener for a VirtualRouter
+   */
+  public static http2(port?: number): VirtualRouterListener {
+    return new VirtualRouterListenerImpl(Protocol.HTTP2, port);
+  }
+
+  /**
+   * Returns a GRPC Listener for a VirtualRouter
+   */
+  public static grpc(port?: number): VirtualRouterListener {
+    return new VirtualRouterListenerImpl(Protocol.GRPC, port);
+  }
+
+  /**
+   * Returns a TCP Listener for a VirtualRouter
+   */
+  public static tcp(port?: number): VirtualRouterListener {
```

Comment:
We should explain what the default port in all of the method docs as well (something like `@param port the optional port of the listener, 8080 by default`).